src/components/ZapWidget/ZapWidget.tsx

```diff
@@ -85,67 +98,416 @@ export function ZapWidget({
             stateMutability: 'view',
             type: 'function',
           },
-        ],
+        ] as const,
         address: (projectData.tokenAddress ||
           projectData.address) as `0x${string}`,
         chainId: projectData.chainId,
         functionName: 'decimals',
       },
-    ],
+    ];
+  }, [
+    projectData.address,
+    projectData.tokenAddress,
+    projectData.chainId,
+    account.address,
+  ]);
+
+  const {
+    data: [
+      { result: depositTokenData } = {},
+      { result: depositTokenDecimals } = {},
+    ] = [],
+    isLoading: isLoadingDepositTokenData,
+    refetch,
+  } = useReadContracts({
+    contracts: contractsConfig,
   });
 
+  useEffect(() => {
+    const initMeeClient = async () => {
+      if (!chain) {
+        console.warn('Chain is undefined, skipping MEE client initialization.');
+        return;
+      }
+
+      // create multichain nexus account
+      // Creates the Biconomy "Multichain Nexus Account", a smart contract account
+      // that orchestrates actions across multiple chains.
+      // See: https://docs.biconomy.io/multichain-orchestration/comprehensive#multichain-nexus-account
+      const oNexusInit = await toMultichainNexusAccount({
+        signer: createWalletClient({
```

Comment:
Might be good to reuse the wagmi configuration we already have (especially for RPC stuff)